### PR TITLE
Begin comment conversion - ensure correct formatting

### PR DESCRIPTION
**Pull request by randypalusz**
Referring to #109

### CODE REVIEW

**Comment by randypalusz** [comment]
I'm curious how to handle commenting overloaded methods. For example, in [`viewport.h`,](https://github.com/UnfoldedInc/deck.gl-native/blob/master/cpp/modules/deck.gl/core/src/viewports/viewport.h) we have two `project` methods:

`auto project(const mathgl::Vector2<double>& lngLat, bool topLeft = true) -> mathgl::Vector2<double>;`

`  auto project(const mathgl::Vector3<double>& lngLatZ, bool topLeft = true) -> mathgl::Vector3<double>;`

Should we have one block of documentation for both methods with options, or two separate documentation blocks?

**Comment by ibgreen** [comment]
Maybe this could be useful: http://www.doxygen.nl/manual/commands.html#cmdoverload

File: cpp/modules/deck.gl/core/src/viewports/viewport.h [hunk 1]
```diff
@@ -90,29 +90,25 @@ class Viewport {
...
-   */
+  /// \brief Projects xyz (possibly latitude and longitude) to pixel coordinates in
+  /// \brief window using viewport projection parameters
+  /// - [longitude, latitude] to [x, y]
+  /// - [longitude, latitude, Z] => [x, y, z]
+  /// \note By default, returns top-left coordinates for canvas/SVG type render
+  ///
+  /// \param lngLat [lng, lat] or [lng, lat, Z]
+  /// \param topLeft Whether projected coords are top left
+  /// \return {Array} - [x, y] or [x, y, z] in top left coords
```
Comment (ilijapuaca) [inline cpp/modules/deck.gl/core/src/viewports/viewport.h:101]: We should get rid of return types (types in general) from our docs, while useful in JS, they are not needed in C++

**Comment by randypalusz** [comment]
Should be mostly converted at this point

File: cpp/modules/deck.gl/core/src/lib/view-manager.h [hunk 2]
```diff
@@ -37,59 +37,56 @@ class ViewManager {
...
-  // Checks each viewport for transition updates
+  /// \brief Checks each viewport for transition updates
   void updateViewStates();
 
-  /** Get a set of viewports for a given width and height
-   * TODO - Intention is for deck.gl to autodeduce width and height and drop
-   * the need for props
-   */
+  /// \brief Get a set of viewports for a given width and height
+  // TODO - Intention is for deck.gl to autodeduce width and height and drop
```
Comment (ilijapuaca) [inline cpp/modules/deck.gl/core/src/lib/view-manager.h:50]: In general, format of `TODO` comments should be `// TODO(email): ...`, cpplint should complain if that's not the case. The email can just be the email of the person that wrote the comment, not necessarily that one that should address it.

In this case though, I believe we can get rid of this comment as it refers to `rect` (as noted next to the method signature) parameter which is not part of this implementation

File: cpp/modules/deck.gl/core/src/shaderlib/project/viewport-uniforms.h [hunk 3]
```diff
@@ -50,15 +50,11 @@ struct ViewportUniforms {
...
- * TODO - Ensure this works with any viewport, not just WebMercatorViewports
- *
- * @param viewport -
- * @return {Float32Array} - 4x4 projection matrix that can be used in shaders
- */
+/// \brief Returns uniforms for shaders based on current projection
+/// \note Includes projection matrix suitable for shaders
+/// \param viewport
+/// \return 4x4 projection matrix that can be used in shaders
+// TODO - Ensure this works with any viewport, not just WebMercatorViewports
```
Comment (ilijapuaca) [inline cpp/modules/deck.gl/core/src/shaderlib/project/viewport-uniforms.h:57]: The format should match the one mentioned above

Comment (randypalusz) [reply cpp/modules/deck.gl/core/src/shaderlib/project/viewport-uniforms.h:57]: I don't think there was an attached email in the original comment. What should be done in a case like this?

File: cpp/modules/deck.gl/core/src/lib/view-manager.h (hunk 2 above)
Comment (randypalusz) [reply cpp/modules/deck.gl/core/src/lib/view-manager.h:50]: Will be removed in next commit

File: cpp/modules/deck.gl/core/src/shaderlib/project/viewport-uniforms.h (hunk 3 above)
Comment (randypalusz) [reply cpp/modules/deck.gl/core/src/shaderlib/project/viewport-uniforms.h:57]: is `// TODO(unknown):` acceptable?

Comment (ilijapuaca) [reply cpp/modules/deck.gl/core/src/shaderlib/project/viewport-uniforms.h:57]: If you know who the original author is, you can use their email, if not you'd just use your own. In this case it was @isaacbrodsky

**Pull request merged**